### PR TITLE
Add parallel tempering (PT) examples

### PR DESCRIPTION
**Pull request by randomir**

### CODE REVIEW

File: examples/parallel-tempering-alt.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+# load a problem
+problem = sys.argv[1]
+with open(problem) as fp:
+    bqm = dimod.BinaryQuadraticModel.from_coo(fp)
+
+n = len(bqm)
+m = len(bqm.quadratic)
+d = 200.0 * m / n / (n - 1)
+print("BQM: {} nodes, {} edges, {:.2f}% density".format(n, m, d))
```
[the remark this answers is not in the document]
Comment (randomir) [reply examples/parallel-tempering-alt.py]: number of existing edges / number of possible edges

[the remark this answers is not in the document]
Comment (randomir) [reply examples/parallel-tempering-alt.py]: See [here](https://en.wikipedia.org/wiki/Dense_graph).

[the remark this answers is not in the document]
Comment (JoelPasvolsky) [reply examples/parallel-tempering-alt.py]: Yeah, it was the 200 that wasn't clear to me.

[the remark this answers is not in the document]
Comment (randomir) [reply examples/parallel-tempering-alt.py]: Long-term, we would probably want to add this utility function/property to BQM: `bqm.density`.

**Pull request merged**